etna/models/tbats.py

```diff
@@ -114,7 +138,15 @@ def predict_components(self, df: pd.DataFrame) -> pd.DataFrame:
         :
             dataframe with prediction components
         """
-        raise NotImplementedError("Prediction decomposition isn't currently implemented!")
+        if self._fitted_model is None or self._freq is None:
+            raise ValueError("Model is not fitted! Fit the model before estimating forecast components!")
+
+        self._check_components()
+
+        raw_components = self._decompose_predict()
```

Comment:
Here we should decompose predictions only on `df`, now you get the decomposition from the whole train dataset